Library/brew.sh

```diff
@@ -201,6 +169,64 @@ EOS
   esac
 fi
 
+if [[ -z "$HOMEBREW_DEVELOPER" ]]
+then
+  unset HOMEBREW_RUBY_PATH
+fi
+
+if [[ -z "$HOMEBREW_RUBY_PATH" && "$HOMEBREW_COMMAND" != "install-vendor" ]]
+then
+  HOMEBREW_VENDOR_RUBY_PATH="$HOMEBREW_LIBRARY/Vendor/Ruby/Versions/Current/bin/ruby"
+  if [[ -x "$HOMEBREW_VENDOR_RUBY_PATH" ]]
+  then
+    HOMEBREW_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_PATH"
+  else
+    if [[ -n "$HOMEBREW_OSX" ]]
+    then
+      HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
+
+      if [[ -d "/System/Library/Frameworks/Ruby.framework/Versions/1.8" ]]
```

Comment:
I'd just sniff the `which ruby` version instead; means users with Rubys installed from elsewhere e.g. `rbenv` can have this work and makes it easier for Linuxbrew too.